kotlin/src/main/kotlin/io/logto/sdk/core/util/TokenUtils.kt

```diff
@@ -2,9 +2,36 @@ package io.logto.sdk.core.util
 
 import io.logto.sdk.core.extension.toIdTokenClaims
 import io.logto.sdk.core.type.IdTokenClaims
+import org.jose4j.jwa.AlgorithmConstraints
+import org.jose4j.jwk.JsonWebKeySet
+import org.jose4j.jws.AlgorithmIdentifiers
 import org.jose4j.jwt.consumer.JwtConsumerBuilder
+import org.jose4j.keys.resolvers.JwksVerificationKeyResolver
 
 object TokenUtils {
+    internal const val ISSUED_AT_RESTRICTIONS_IN_SECONDS = 60
+
+    fun verifyIdToken(
+        idToken: String,
+        clientId: String,
+        issuer: String,
+        jwks: JsonWebKeySet,
+    ) {
+        JwtConsumerBuilder().apply {
+            setRequireSubject()
+            setRequireExpirationTime()
+            setRequireIssuedAt()
+            setExpectedIssuer(issuer)
+            setExpectedAudience(clientId)
+            setIssuedAtRestrictions(ISSUED_AT_RESTRICTIONS_IN_SECONDS, ISSUED_AT_RESTRICTIONS_IN_SECONDS)
+            setJwsAlgorithmConstraints(
+                AlgorithmConstraints.ConstraintType.PERMIT,
+                AlgorithmIdentifiers.RSA_USING_SHA256,
+            )
```

Comment:
we should allow all jwk supported algorithms like `EC512` since it's better and we may have a change to use?

Reply:
Updated.
Yes, I add this for we have a previous convention that says we only use S256 in v1.
It seems we don't need this constraint anymore.
This jose lib supports all algorithms we need.
FYI: [source code about supported algorithms](https://bitbucket.org/b_c/jose4j/src/ff6d6c2ee7b94ca5e1ee9c17b4647938110bd07d/src/main/java/org/jose4j/jwa/AlgorithmFactoryFactory.java#lines-54:115)